File: frontend/src/api/statements.ts
import { apiClient } from "./client";
import type { StatementFilters, StatementPage } from "../types/api";

export async function listStatements(
  filters: StatementFilters,
): Promise<StatementPage> {
  const params: Record<string, string | number> = {
    page: filters.page,
    page_size: filters.page_size,
  };
  if (filters.date_from) params.date_from = filters.date_from;
  if (filters.date_to) params.date_to = filters.date_to;
  if (filters.cedente_id) params.cedente_id = filters.cedente_id;
  if (filters.currency_code) params.currency_code = filters.currency_code;
  if (filters.status) params.status = filters.status;

  const { data } = await apiClient.get<StatementPage>(
    "/api/v1/reports/statements",
    { params },
  );
  return data;
}

File: backend/app/schemas/statements.py
import uuid
from dataclasses import dataclass
from datetime import date, datetime
from decimal import Decimal

from pydantic import BaseModel


@dataclass(frozen=True)
class StatementFilters:
    date_from: date | None = None
    date_to: date | None = None
    cedente_id: uuid.UUID | None = None
    payment_currency_code: str | None = None
    status: str | None = None


class StatementItem(BaseModel):
    id: uuid.UUID
    created_at: datetime
    cedente_name: str
    cedente_document: str
    receivable_type: str
    face_value: Decimal
    present_value: Decimal
    currency_code: str
    payment_currency_code: str
    exchange_rate_used: Decimal | None
    status: str


class StatementPage(BaseModel):
    page: int
    page_size: int
    total: int
    items: list[StatementItem]
